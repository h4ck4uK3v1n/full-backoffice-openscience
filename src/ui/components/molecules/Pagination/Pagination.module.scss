.pagination {
    align-items: center;
    background: transparent;
    display: flex;
}
