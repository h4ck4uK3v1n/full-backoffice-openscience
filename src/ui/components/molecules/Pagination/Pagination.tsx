import { useState } from 'react';
import styles from './Pagination.module.scss';
import LabelPage from '../../atoms/Label-Page/Page-Label';
import ChevronButton from '../../atoms/ButtonChevron/Chevron-Button';
import type { PaginationProps } from './types/IProps';

import left from '../../../../assets/icons/angle-left.svg?raw';
import right from '../../../../assets/icons/angle-right.svg?raw';

function Pagination({ currentPage = 1, totalPages, onClick }: PaginationProps) {
  // Controla el estado de la página actual internamente
  const [current, setCurrent] = useState(currentPage);

  // Manejador para el botón de retroceder página
  const handlePrevClick = () => {
    if (current > 1) {
      setCurrent(current - 1); // Actualiza la página actual restando 1
      if (onClick) onClick(current - 1); // Llama a onClick con la nueva página
    }
  };

  // Manejador para el botón de avanzar página
  const handleNextClick = () => {
    if (current < totalPages) {
      setCurrent(current + 1); // Actualiza la página actual sumando 1
      if (onClick) onClick(current + 1); // Llama a onClick con la nueva página
    }
  };

  return (
    <div className={`${styles.pagination}`}>
      <ChevronButton icon={left} onClick={handlePrevClick} />
      <LabelPage currentPage={current} totalPages={totalPages} />
      <ChevronButton icon={right} onClick={handleNextClick} />
    </div>
  );
}

export default Pagination;
