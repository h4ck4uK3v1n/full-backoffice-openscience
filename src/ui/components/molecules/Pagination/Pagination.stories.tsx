import type { StoryObj } from '@storybook/react';
import Pagination from './Pagination';

const meta = {
  title: 'ui/components/molecules/Pagination',
  component: Pagination,
  argTypes: {
    onClick: {
      action: 'clicked',
    },
  },
};

export default meta;

type Story = StoryObj<typeof Pagination>;

export const Default: Story = {
  args: {
    currentPage: 1,
    totalPages: 5,
  },
};

export const MiddlePage: Story = {
  args: {
    currentPage: 3,
    totalPages: 5,
  },
};

export const LastPage: Story = {
  args: {
    currentPage: 5,
    totalPages: 5,
  },
};
