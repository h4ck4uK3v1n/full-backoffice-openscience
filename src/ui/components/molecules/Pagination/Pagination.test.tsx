import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Pagination from './Pagination';

describe('Pagination', () => {
  it('renders correctly', () => {
    const { container } = render(<Pagination currentPage={1} totalPages={10} onClick={() => {}} />);
    expect(container.innerHTML).toMatchSnapshot();
  });
});
