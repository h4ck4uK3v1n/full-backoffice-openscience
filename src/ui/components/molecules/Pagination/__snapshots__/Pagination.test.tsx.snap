// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pagination > renders correctly 1`] = `"<div class="_pagination_e19fa2"><button class="_button-chevron_f3de84"><span data-testid="icon-chevron" class="_icon-chevron_f3de84"><svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M15.2803 7.28033C15.5732 6.98744 15.5732 6.51256 15.2803 6.21967C14.9874 5.92678 14.5126 5.92678 14.2197 6.21967L9.21967 11.2197C8.92678 11.5126 8.92678 11.9874 9.21967 12.2803L14.2197 17.2803C14.5126 17.5732 14.9874 17.5732 15.2803 17.2803C15.5732 16.9874 15.5732 16.5126 15.2803 16.2197L10.8107 11.75L15.2803 7.28033Z" fill="#131313"></path></svg></span></button><div class="_indicator_181dd7">Page 1 of 10</div><button class="_button-chevron_f3de84"><span data-testid="icon-chevron" class="_icon-chevron_f3de84"><svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M9.21967 16.2197C8.92678 16.5126 8.92678 16.9874 9.21967 17.2803C9.51256 17.5732 9.98744 17.5732 10.2803 17.2803L15.2803 12.2803C15.5732 11.9874 15.5732 11.5126 15.2803 11.2197L10.2803 6.21967C9.98744 5.92678 9.51256 5.92678 9.21967 6.21967C8.92678 6.51256 8.92678 6.98744 9.21967 7.28033L13.6893 11.75L9.21967 16.2197Z" fill="#131313"></path></svg></span></button></div>"`;
